### DISCUSSION TITLE
Feature request - cast json with missing keys to struct with nulls

### DISCUSSION THREAD

**Discussion by eaubin** [duckdb/duckdb#10517] [Ideas]
It's very convenient to cast json to structs

```sql
CREATE TYPE xy AS STRUCT(x INT, y INT);

SELECT '{"x":1,"y":2}'::JSON::xy
```

Of course structs require values for all fields, so casting an incomplete json object to a struct fails

```sql
SELECT '{"x":1}'::JSON::xy
-- Conversion Error: Object {"x":1} does not have key "y"
```

To work around this you can create a macro that checks for the presence of each struct key and when present cast the field to the right type or use null.

Could this kind loose cast operator be created generically? Is there a way in SQL to access the fields and types of a struct?

**Comment by alexthe2** [comment]
can you create this as an issue?

**Reply by eaubin** [reply]
I believe the devs want to keep the feature requests in discussions
